staging/src/k8s.io/apiserver/pkg/authentication/request/bearertoken/bearertoken.go

```diff
@@ -48,6 +50,10 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 
 	// Empty bearer tokens aren't valid
 	if len(token) == 0 {
+		// The space before the token case
+		if len(parts) == 3 {
+			klog.Warning("WARNING: the support for spaces before the token will be removed shortly")
```

Comment:
A kube-apiserver log warning is not visible to the person making the request, who is the one that is likely to be confused by lack of authentication and is the one who needs to make a change. Also, we haven't decided that this type of request will turn into an error in the future, so don't include that in the message.

This should be something like this:
```go
warning.AddWarning(req.Context(), "", "the provided Authorization header contains extra space before the bearer token, and is ignored")
```

Reply:
Sure! I just changed the text slightly: `the provided Authorization header contains extra space before the bearer token, which is ignored`. I hope it's OK.